src/resultsdisassemblypage.cpp

```diff
@@ -186,6 +232,16 @@ void ResultsDisassemblyPage::showDisassembly(const DisassemblyOutput& disassembl
     m_disassemblyModel->clear();
     m_sourceCodeModel->clear();
 
+#if KF5SyntaxHighlighting_FOUND
+    auto sourceCodeDefinition = m_repository->definitionForFileName(disassemblyOutput.mainSourceFileName);
+
+    ui->assemblyComboBox->setCurrentText(QStringLiteral("GNU Assembler"));
```

Comment:
this is the wrong place to do that

Reply:
why? this needs to be set if a file gets opened, in case the user changed something

Reply:
the problem is that you have to set it twice, manually. that's wrong. I think here it should only be needed to set the definition once on the respective model which should then emit a signal which lets the ui adapt to select the corresponding item